cl/citations/tests.py

```diff
@@ -1085,3 +1093,267 @@ def _print_failed_cases(
+                        text='Finding that forfeitures are fines "if they constitute punishment for an offense"',
+                        id=4,
+                        score=0,
+                    ),
+                ],
+                [
+                    DummyParenthetical(
+                        text="Despite differing facts emphasized by the majority and dissent, the majority held that “Respondent’s crime was solely a reporting offense”",
+                        id=5,
+                        score=0,
+                    ),
+                ],
+                [
+                    DummyParenthetical(
+                        text="“[D]espite the differences between restitution and a traditional fine, restitution still implicates the prosecutorial powers of government[.]”",
+                        id=6,
+                        score=0,
+                    ),
+                ],
+            ),
+            (
+                [
+                    DummyParenthetical(
+                        text="Finding valid a federal law criminalizing the destruction or mutilation of a draft registration against a First Amendment challenge",
+                        id=10,
+                        score=0,
+                    ),
+                ],
+                [
+                    DummyParenthetical(
+                        text="Applying medium scrutiny test to state action having an incidental effect on right to free expression ",
+                        id=11,
+                        score=0,
+                    )
+                ],
+            ),
+            (
+                [
+                    DummyParenthetical(
+                        text="The loss of First Amendment freedoms, for even minimal period of time, unquestionably constitutes irreparable injury.",
+                        id=11,
+                        score=0,
+                    ),
+                    DummyParenthetical(
+                        text="The loss of First Amendment freedoms, for even minimal period of time, unquestionably constitutes irreparable injury.",
+                        id=12,
+                        score=0,
+                    ),
+                    DummyParenthetical(
+                        text="The loss of First Amendment freedoms, for even minimal period of time, unquestionably constitutes irreparable injury.",
+                        id=13,
+                        score=0,
+                    ),
+                ],
+                [
+                    DummyParenthetical(
+                        text=" Holding public employees could not be fired because of their politics unless they held “policymaking” or “confidential” positions ",
+                        id=14,
+                        score=0,
+                    ),
+                ],
+            ),
+        ]
+        for i, groups in enumerate(expected_groups):
+            with self.subTest(f"Testing {groups} are grouped correctly.", i=i):
```

Comment:
Does this make ridiculous log lines with the entire group printed, since you put `{groups}` in there?

Reply:
No --- I defined the `__str__` and `__repr__` of `groups` to only print `DummyParenthetical.id`, which keeps the printed length of `groups` short, and it's helpful to see group info in debug. The consequence is that you'd have to go back to the tests file to see what text corresponded to a particular ID. But my guess was that if tests are failing, you'd be doing something like that anyway, and it was the most printable way I could think of to communicate the most information I could about the test.

Reply:
My fault: removed `__str__` and `__repr__` definitions without realizing they were being implicitly relied upon here